### PR TITLE
OTP

### PR DESCRIPTION
**Pull request by andikabahari**
Belum:
- Send pin OTP ke email user (cek service OTP)

Sudah:
- Request pin OTP
- Verifikasi pin OTP

### CODE REVIEW

File: services/otp_service/otp_service.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+	otp, err := s.otpRepository.FindByPinAndUserID(payload.Pin, user.ID)
+	if err != nil {
+		return nil, errors.New("invalid pin")
+	}
+
+	if time.Now().After(otp.ExpiredAt) {
+		return nil, errors.New("otp expired")
+	}
+
+	s.otpRepository.DeleteByID(otp.ID)
```
Comment (j4ceu) [inline services/otp_service/otp_service.go:76]: ditambahin err := nya di bagian line ini dik

**Review by j4ceu — CHANGES_REQUESTED** [review]
Sama tolong tambahin boolean isUsed di modelnya dik, thank uu

File: services/otp_service/otp_service.go (hunk 1 above)
Comment (andikabahari) [reply services/otp_service/otp_service.go]: - nambah field is_used
- handle error
- delete diganti jadi update is_used = true

Comment (andikabahari) [reply services/otp_service/otp_service.go]: di sini [853018d](https://github.com/Kelompok4-LoyaltyPointAgent/Backend/pull/8/commits/853018d3640fe63c623cf61d5a5fa0c59be59fc0)

**Pull request merged**